app/controllers/users_controller.rb

```diff
@@ -1,5 +1,18 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action { authorize @user || User}
+
+  def feed
+    
+  end
+
+  def discover
+    
+  end
+
+  def liked
+    
+  end
```

Comment:
We don't need to define methods for feed, discover, liked. Because the view name matches the action, Rails knows what view to render.